velox/connectors/hive/HiveConnector.cpp

```diff
@@ -347,17 +347,57 @@ HiveDataSource::HiveDataSource(
 }
 
 namespace {
+bool applyPartitionFilter(
+    TypeKind kind,
+    std::string partitionValue,
+    common::Filter* filter) {
+  switch (kind) {
+    case TypeKind::BIGINT:
+    case TypeKind::INTEGER:
+    case TypeKind::SMALLINT:
+    case TypeKind::TINYINT: {
+      return applyFilter(*filter, folly::to<int64_t>(partitionValue));
+    }
+    case TypeKind::REAL:
+    case TypeKind::DOUBLE: {
+      return applyFilter(*filter, folly::to<double>(partitionValue));
+    }
+    case TypeKind::BOOLEAN: {
+      return applyFilter(*filter, folly::to<bool>(partitionValue));
+    }
+    case TypeKind::VARCHAR: {
+      return applyFilter(*filter, partitionValue);
+    }
+    default:
+      VELOX_CHECK(false, "Bad partitionKey type to filter");
+      break;
+  }
+}
+
 bool testFilters(
     common::ScanSpec* scanSpec,
     dwio::common::Reader* reader,
-    const std::string& filePath) {
+    const std::string& filePath,
+    const std::unordered_map<std::string, std::optional<std::string>>&
+        partitionKey,
+    std::unordered_map<std::string, std::shared_ptr<HiveColumnHandle>>&
+        partitionKeysHandle) {
   auto totalRows = reader->numberOfRows();
   const auto& fileTypeWithId = reader->typeWithId();
   const auto& rowType = reader->rowType();
   for (const auto& child : scanSpec->children()) {
     if (child->filter()) {
       const auto& name = child->fieldName();
       if (!rowType->containsChild(name)) {
+        if (child->isPartitionKey()) {
```

Comment:
Actually I am wondering do we really need to mark it as partition key in `ScanSpec`?  If we can find the `name` in `partitionKey` here, we can already decide that it is a partition key and call `applyPartitionFilter` on it, no?

Reply:
I think it makes sense. I'll make some changes.